huaweicloud/services/workspace/resource_huaweicloud_workspace_desktop_user_batch_attach.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	result := make([]interface{}, 0, len(userInfos))
+	for _, userInfo := range userInfos {
+		result = append(result, utils.RemoveNil(map[string]interface{}{
+			"type":       utils.ValueIgnoreEmpty(utils.PathSearch("type", userInfo, "USER")),
+			"user_id":    utils.ValueIgnoreEmpty(utils.PathSearch("user_id", userInfo, "")),
+			"user_name":  utils.ValueIgnoreEmpty(utils.PathSearch("user_name", userInfo, "")),
+			"user_group": utils.ValueIgnoreEmpty(utils.PathSearch("user_group", userInfo, "")),
+		}))
+	}
+
+	return result
+}
+
+func buildDesktopUserBatchAttachDesktopsBodyParams(desktops []interface{}) []interface{} {
+	if len(desktops) < 1 {
+		return nil
+	}
+
+	result := make([]interface{}, 0, len(desktops))
+	for _, desktop := range desktops {
+		desktopMap := map[string]interface{}{
+			"desktop_id":    utils.ValueIgnoreEmpty(utils.PathSearch("desktop_id", desktop, "")),
+			"computer_name": utils.ValueIgnoreEmpty(utils.PathSearch("computer_name", desktop, "")),
+			"user_name":     utils.ValueIgnoreEmpty(utils.PathSearch("user_name", desktop, "")),
+			"user_group":    utils.ValueIgnoreEmpty(utils.PathSearch("user_group", desktop, "")),
+			"user_email":    utils.ValueIgnoreEmpty(utils.PathSearch("user_email", desktop, "")),
+			"is_clear_data": utils.PathSearch("is_clear_data", desktop, true),
+		}
+
+		if v, ok := utils.PathSearch("attach_user_infos", desktop, []interface{}{}).([]interface{}); ok && len(v) > 0 {
+			desktopMap["attach_user_infos"] = buildDesktopUserBatchAttachUserInfosBodyParams(v)
+		}
+
+		result = append(result, utils.RemoveNil(desktopMap))
+	}
+
+	return result
+}
+
+func buildDesktopUserBatchAttachBodyParams(d *schema.ResourceData) map[string]interface{} {
+	body := map[string]interface{}{}
+
+	if v, ok := d.GetOk("desktops"); ok {
+		body["desktops"] = buildDesktopUserBatchAttachDesktopsBodyParams(v.([]interface{}))
+	}
+
+	return body
+}
+
+func resourceDesktopUserBatchAttachCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/desktops/attach"
+	)
+
+	client, err := cfg.NewServiceClient("workspace", region)
+	if err != nil {
+		return diag.Errorf("error creating Workspace client: %s", err)
+	}
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json",
+		},
+		JSONBody: buildDesktopUserBatchAttachBodyParams(d),
+	}
+
+	resp, err := client.Request("POST", createPath, &opt)
+	if err != nil {
+		return diag.Errorf("error creating desktop user batch attach: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	jobId := utils.PathSearch("job_id", respBody, "")
+
+	if jobId != nil && jobId.(string) != "" {
```

Comment:
set the default value as empty string mean this job ID could never be nil
```suggestion
	if jobId.(string) != "" {
```